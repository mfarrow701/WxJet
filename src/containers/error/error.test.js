import React from 'react';
import {shallow} from 'enzyme';
import Error from './error';

describe('Error container', () => {
    let component;

    const renderWrapper = () => {
        component = shallow(<Error/>);
    };

    beforeEach(() => {
        renderWrapper();
    });

    it('should render the error container after load', () => {
        expect(component).toMatchSnapshot();
        expect(component.find('.Error h1').text()).toEqual('No message specified');
    });

    it('should render the error container with a custom message', () => {
        component.setProps({message: 'Page not found'});
        expect(component.find('.Error h1').text()).toEqual('Page not found');
    });
});
