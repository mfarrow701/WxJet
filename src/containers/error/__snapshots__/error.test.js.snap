// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error container should render the error container after load 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Error />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        No message specified
</h1>,
      "className": "Error",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "No message specified",
      },
      "ref": null,
      "rendered": "No message specified",
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          No message specified
</h1>,
        "className": "Error",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "No message specified",
        },
        "ref": null,
        "rendered": "No message specified",
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
